### ISSUE TITLE
Favicon fetches can cause access control bypass

### ISSUE THREAD

**Issue by robertknight** [hypothesis/viahtml#77]
TL;DR - Browser requests for viahtml's favicon can set a `via.sec` cookie which allows authentication checks to be bypassed on subsequent requests.

**Steps to reproduce:**

1. Open a new Private Browsing tab (this step ensures there are no cached cookies for viahtml)
2. Go to /proxy/https://www.theguardian.com/world/2021/jan/14/brazil-manaus-amazonas-covid-coronavirus on an instance of ViaHTML that is configured to require authentication (currently https://qa-viahtml3.hypothes.is/proxy/https://www.theguardian.com/world/2021/jan/14/brazil-manaus-amazonas-covid-coronavirus is configured this way)
3. Reload the page

**Expected result:**

Step 2 should fail with a 401 response. Step 3 should also fail with a 401 response.

**Actual result:**

Step 3 succeeds and loads the site. The reason for this is that even though Step 2 results in a 401 response, the browser ends up fetching a favicon from https://qa-viahtml3.hypothes.is/favicon.ico. The favicon request has `Referer` and `Sec-Fetch-Site` headers which identify the request as coming from the original proxied page.

The response to the favicon fetch request sets a `via.sec` cookie which causes subsequent requests for any page to succeed.
